### PR TITLE
Provide a flag to pass environment variables at runtime

### PR DESCRIPTION
**Pull request by rprimet**
Submitted in partial fulfillment for #32 

I'm wet-behind-the-ears when it comes to Python; I'd appreciate constructive criticism :-)

### CODE REVIEW

**Comment by betatim** [comment]
Yay! Just had a quick look at your PR and I think you are lowering expectations with your "wet-behind-the-ears" comment. Looks pretty solid to me! I will take a closer look tomorrow morning/when travis is done.

One thing off the top of my head: we had some problem with the repo path and the command-to-run getting in the way of each other. Basically argparse parsing malarkey :-/ So yeah I'd poke around there a bit. Otherwise this looks solid.

**Comment by rprimet** [comment]
So... CI fails and I'm not sure why. Any help? Otherwise I'll just close the PR.

**Comment by betatim** [comment]
To find a test that is failing (in the huge amount of output in the log) I search for "FAILED". In this case it gives me https://travis-ci.org/jupyter/repo2docker/jobs/326160395#L1192 looks like the failure isn't your fault. It mentions that it can't resolve a package called wheel, with a certain version. 

I get similar error locally when trying to run `conda install "wheel=0.30.0"`, seems the latest version conda wants to install is 0.29.0. However conda-forge has 0.30.0.

**Comment by betatim** [comment]
I'd go investigate the `repo2docker/buildpacks/conda/environment.py-3.6.frozen.yml` and friends to understand why wheel 0.30.0 is unresolved despite existing on conda-forge and that channel being selected. @minrk recently did some work on that so you could also ask him for advice.

File: repo2docker/app.py [hunk 1]
```diff
@@ -297,6 +297,14 @@ def get_argparser(self):
             help='Username of the primary user in the image',
         )
 
+        argparser.add_argument(
+            '--env', '-e',
+            dest='environment',
+            nargs='+',
```
Comment (yuvipanda) [inline repo2docker/app.py]: action='append' is probably better here than nargs='+'

**Comment by rprimet** [comment]
@yuvipanda : changed in d5d9b89 but there is a new CI failure -- I can't tell if it's spurious or not.

**Comment by rprimet** [comment]
The change to `action='append'` is done

**Comment by rprimet** [comment]
Cool, thanks for the merge!

**Comment by betatim** [comment]
🎉 !

**Pull request merged**